### PR TITLE
Fixes #1788: Show Admin api call, reducer and action implemented [Fully functional]

### LINKED ISSUE #1788: Implement show admin API and call it after successful login
**Actual Behaviour**

Currently, the api is being called via $ and is being called when static app bar mounts which is not ideal as everytime the route changes, the api is being called again.

**Expected Behaviour**

The changes I propose are:
- [ ] Define the api calls in api/index.js and remove $ dependency.
- [ ] Calling the api only when a successful login happens.
- [ ] Hydrate the store accordingly.

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by AakashMallik**
Fixes #1788

Changes: The changes I have made are as follows:
- [x] Define api call in api/index.js
- [x] Define action for getting isAdmin value
- [x] Define reducer for isAdmin
- [x] Use it in index.js and login.react.js 
- [x] Fix merge conflict

Surge Deployment Link: https://pr-1790-fossasia-susi-skill-cms-new.surge.sh

Note: CI was not tearing down the old link and probably that is why the link was not updating. Here is a new link.

### CODE REVIEW

**Comment by AakashMallik** [comment]
This PR is dependent on https://github.com/fossasia/susi_skill_cms/pull/1789, so I would finalize this when I am finished with that PR.

File: src/index.js [hunk 1]
```diff
@@ -93,6 +93,7 @@ class App extends React.Component {
   componentDidMount = () => {
     const { actions } = this.props;
     actions.getApiKeys();
+    actions.getAdmin();
```
Comment (fragm3) [inline src/index.js]: getAdmin should only be called when the user is logged in, put a check here @AakashMallik

Comment (AakashMallik) [reply src/index.js]: @fragm3 Will do. Btw, I noticed that in your code in Susi.Chat, you passed access_token explicitly in the getAdminApi. Was that needed coz ajax.js would put the "access_token" value anyway. Let me know as I have a pending PR, so I can make the changes before finalizing it.

**Comment by AakashMallik** [comment]
@fragm3 Let me know if this is okay

File: src/index.js (hunk 1 above)
Comment (fragm3) [reply src/index.js]: @AakashMallik we will need to pass the access token separately on login, as the access token is not set till then(async in nature)

**Comment by AakashMallik** [comment]
@akshatnitd Sir it did not work after refresh because @fragm3 's PR https://github.com/fossasia/susi_skill_cms/pull/1832 has not been merged, so we don't have the access_token from the cookies as they would be lost after refresh and @fragm3 's PR hydrates the store which has not been merged. If his PR is merged, my PR will work on refresh too.

**Comment by Orbiter** [comment]
https://github.com/fossasia/susi_skill_cms/pull/1832 is merged

**Comment by Orbiter** [comment]
Please resolve conflicts

**Comment by AakashMallik** [comment]
@akshatnitd @Orbiter Now it should work flawlessly.

**Comment by AakashMallik** [comment]
@akshatnitd Sir I tried redeploying the link but oddly, the code works fine locally but when uploaded on surge, the cookies get flushed on every refresh. Interestingly, this is not happening on the locally deployed code. @fragm3 @samagragupta @anshumanv  I you are free, could you pull my code on your machine, try this out and give me a confirmation? I would be grateful.

**Comment by AakashMallik** [comment]
@akshatnitd Sir, I tried that, but some how, the feature fails to work on surge. Not sure why. Locally it works all fine, but after deploying it, the cookies get deleted after a refresh and that is why the GET_ADMIN action is not being called there. I would like someone to give me a confirmation that it is woking locally for them.

**Comment by fragm3** [comment]
@AakashMallik I'll try it locally

**Comment by fragm3** [comment]
@AakashMallik There is some problem, after loging in, if I refresh all the user app data is lost and it again goes back to not-login state, on localhost.

**Comment by samagragupta** [comment]
@AakashMallik I have the same problem as with @fragm3 .

**Comment by AakashMallik** [comment]
@fragm3 @samagragupta Thanks, folks... Just checked my code and realized I had made a fundamental mistake. I have updated the code, now the surge link should work.

**Comment by AakashMallik** [comment]
@akshatnitd Sir, I have checked the surge link, and now it is working fine. I had commited a blunder. Fixed it.

**Pull request merged**